doc/manual/getting_started_core.rst

```diff
@@ -341,17 +348,17 @@ Subkernels can call other kernels and subkernels. For a more complex example: ::
             assert result == 4
             self.pulse_ttl(20)
 
-Without the preload, the delay after the core reset would need to be longer. It's still an operation that can take some time, depending on the connection. Notice that the method ``pulse_ttl()`` can be also called both within a subkernel, and on its own. 
+Without the preload, the delay after the core reset would need to be longer. The operation may still take some time, depending on the connection. Notice that the method ``pulse_ttl()`` can be called both within a subkernel and on its own. 
 
-In general, subkernels do not have to be awaited, but awaiting is required to retrieve returned values and exceptions.
+It is not necessary for subkernels to always be awaited, but awaiting is required to retrieve returned values and exceptions.
```

Comment:
It's propagated to the master, but the exception isn't raised by the await function within the main kernel, rather it's sent to the host directly.

Reply:
I believe it was a simplification at the time: https://github.com/m-labs/artiq/blob/master/artiq/firmware/runtime/session.rs#L694

Serving the raw exception data from comm core was simpler than returning it to the kernel and handling it there. As one little thing from the whole subkernel subsystem, it was left as-is. But I'm quite sure it would be not too difficult to do it properly.